x-pack/plugins/apm/server/lib/alerts/chart_preview/helper.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import datemath from '@elastic/datemath';
+import Boom from '@hapi/boom';
+import { i18n } from '@kbn/i18n';
+
+const BUCKET_SIZE = 20;
+
+export function getIntervalAndTimeRange({
+  windowSize,
+  windowUnit,
+}: {
+  windowSize: number;
+  windowUnit: string;
+}) {
+  const start = datemath
+    .parse(`now-${windowSize * BUCKET_SIZE}${windowUnit}`)
+    ?.valueOf();
+
+  if (!start) {
+    throw Boom.internal(
+      i18n.translate('xpack.apm.api.alert.chartPreview.invalidWindow', {
+        defaultMessage: `Invalid window {windowSize} {windowUnit}`,
+        values: { windowSize, windowUnit },
+      })
+    );
+  }
```

Comment:
This should only happen if `windowUnit` is invalid, right? Perhaps you can instead validate `windowUnit` with io-ts at the api layer?